docs/_includes/options/data/selection-access.html

```diff
@@ -0,0 +1,50 @@
+<section>
+  <h2 id="selection-api-access">
+    How to programmatically access a selection data?
+  </h2>
+
+  <p>
+    There are few ways to programmatically access the selection data. Calling <code>select2('data')</code> will return the JavaScript array of an objects representing the current selection. Each object will have properties/values which was in the source data objects passed through <code>processResults</code> and <code>templateResult</code> functions (as in <a href="#data">Loading data from an array</a> and <a href="#ajax">Connecting to a remote data source</a>).
+  </p>
+
+{% highlight js linenos %}
+$('select').select2('data');
+{% endhighlight %}
+
+  <p>
+    As Select2 uses the HTML <code>&lt;SELECT&gt;</code> element to store the selection result, the selection data are represented by <code>&lt;OPTION&gt;</code> elements and can be accessed in the plain jQuery/DOM manner. The resulting elements will have properties/values from the source data objects, stored by the means of jQuery <code>data()</code> method and accessible by key <code>'data'</code>:
+  </p>
+
+{% highlight js linenos %}
+// Retrieve source data object's data of the first selected element
+$('select').find(':selected').data('data');
```

Comment:
The locally cached `data` key on the `<option>` is an implementation detail that is definitely private API. It's actually something we're planning on moving away from in the future.